src/day8.scala

```diff
@@ -0,0 +1,130 @@
+// using scala 3.0.2
+
+package day8
+
+import scala.util.Using
+import scala.io.Source
+
+import Segment.*
+import Digit.*
+
+@main def part1: Unit =
+  println(s"The solution is ${part1(readInput())}")
+
+@main def part2: Unit =
+  println(s"The solution is ${part2(readInput())}")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day8"))(_.mkString)
+
+type Segments = Seq[Segment]
+
+def part1(input: String) =
+
+  def getDisplay(line: String) = line.split("[|]")(1).trim
+
+  val uniqueDigits =
+    for
+      display <- input.linesIterator.map(getDisplay)
+      segments <- display.split(" ")
+      uniqueDigit <- parseUniqueDigit(segments)
+    yield
+      uniqueDigit
+
+  uniqueDigits.length
```

Comment:
`length` is usually reserved for `Array`s and `String`s, for which it is the native method name. Consider using `size` instead on general Scala collections.